src/types/tdigest.cc

```diff
@@ -0,0 +1,439 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+/*
+This implementation follows apache arrow.
+refer to https://github.com/apache/arrow/blob/27bbd593625122a4a25d9471c8aaf5df54a6dcf9/cpp/src/arrow/util/tdigest.cc
+*/
+
+#include "tdigest.h"
+
+#include <fmt/format.h>
+#include <glog/logging.h>
+
+#include <algorithm>
+#include <iterator>
+#include <memory>
+#include <queue>
+
+#include "status.h"
+
+namespace {
+// scale function K0: linear function, as baseline
+struct ScalerK0 {
```

Comment:
Seems this is unused?

Reply:
Hi @mapleFU ,

This implement follows Apache arrow. 
I find it is also [unused in arrow](https://github.com/apache/arrow/blob/27bbd593625122a4a25d9471c8aaf5df54a6dcf9/cpp/src/arrow/util/tdigest.cc#L53).

I'm not sure whether I need to keep it and try to follow the original implementation, so I kept it in code.
I guess it is reserved for scale function's benchmark.

Maybe it can be removed until we try to do some benchmark for the scale function. 😄

Best Regards,
Edward